### PR TITLE
fix mData NULL deref in rtString::append

### PR DESCRIPTION
**Pull request by ptrekels**

### CODE REVIEW

File: src/rtString.cpp [hunk 1]
```diff
@@ -91,10 +91,14 @@ void rtString::term()
 
 void rtString::append(const char* s) 
 {
-  size_t sl = strlen(s);
-  size_t dl = strlen(mData);
```
Comment (dwrobel) [inline src/rtString.cpp:95]: Wasn't it enough to modify that to:  `size_t dl = mData ? strlen(mData) : 0`

Comment (ptrekels) [reply src/rtString.cpp:95]: indeed but now the value of s is also checked, to get it more monkey proof

File: src/rtString.cpp [hunk 2]
```diff
@@ -91,10 +91,14 @@ void rtString::term()
...
-  mData = (char*)realloc((void*)mData, dl+sl+1);
-  strcpy(mData+dl, s);
+  if (mData) {
+    size_t sl = strlen(s);
```
Comment (dwrobel) [inline src/rtString.cpp:95]: This doesn't check `s`.

Comment (ptrekels) [reply src/rtString.cpp:95]: indeed.. will rework

**Pull request closed without merging**